### PR TITLE
adds ticket 3 (getCategories)

### PR DESCRIPTION
**Pull request by JJW-27**

### CODE REVIEW

File: __tests__/app.tests.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        expect(res.body.categories.length).toBe(4);
+        res.body.categories.forEach(category => {
+          expect(category).toMatchObject({
+            slug: expect.any(String),
+            description: expect.any(String),
+          });
+        });
+      });
+  });
+});
```
Comment (HannahBurks) [inline __tests__/app.tests.js:31]: Please add a test for the 404 Error also :)

File: app.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const express = require('express');
+
+const { getCategories } = require('./controllers/controllers.js');
+
+const app = express();
+
+app.use(express.json());
```
Comment (HannahBurks) [inline app.js:7]: We are not using this yet, so please remove for the time being :)

File: app.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  res.status(404).send({ msg: 'Path not found' });
+});
+
+app.use((err, req, res, next) => {
+  res.status(400).send({msg: err.msg})
+  } 
+);
+
+app.use((err, req, res)=> {
+  console.log(err)
```
Comment (HannahBurks) [inline app.js:21]: Let's add status 500 error handling here :)

File: __tests__/app.tests.js (hunk 1 above)
Comment (JJW-27) [reply __tests__/app.tests.js:31]: THank you :), then am I okay to merge?

File: app.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+app.use(express.json());
+
+app.get('/api/categories', getCategories);
+
+app.all('/*', (req, res, next) => {
+  res.status(404).send({ msg: 'Path not found' });
+});
+
+app.use((err, req, res, next) => {
+  res.status(400).send({msg: err.msg})
```
Comment (HannahBurks) [inline app.js:16]: We are not using the 400 error currently, so let's remove this for the time being including the associated catch block :)

**Review by HannahBurks — CHANGES_REQUESTED** [review]
Hey Joe, nice work, a few changes to be made before merging. Please send another pull request once these have been corrected 😃

**Pull request closed without merging**